osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/endpoints/SecretManagementEndpoint.java

```diff
@@ -0,0 +1,138 @@
+
+        log.info("Handling incoming SOAP request 'getSecretsRequest' for device {}", request.getDeviceId());
+
+        if (log.isDebugEnabled()) {
+            log.debug(getSecretsRequestToString(request));
+        }
+
+        try {
+            GetSecretsResponse response = new GetSecretsResponse();
+
+            SecretTypes soapSecretTypes = request.getSecretTypes();
+
+            if (soapSecretTypes == null) {
+                throw new TechnicalException("Missing input: secret types");
+            }
+
+            List<SecretType> secretTypeList = converter.convertToSecretTypes(soapSecretTypes);
+            List<TypedSecret> typedSecrets = secretManagementService.retrieveSecrets(request.getDeviceId(),
+                    secretTypeList);
+
+            TypedSecrets soapTypedSecrets = converter.convertToSoapTypedSecrets(typedSecrets);
+
+            response.setTypedSecrets(soapTypedSecrets);
+            response.setResult(OsgpResultType.OK);
+
+            log.trace(response.toString());
+
+            return response;
+
+        } catch (Exception e) {
+            if (log.isTraceEnabled()) {
+                e.printStackTrace();
+            }
+            throw new TechnicalException(ComponentType.SHARED, e.getMessage());
+        }
+    }
+
+    @PayloadRoot(namespace = NAMESPACE_URI, localPart = "storeSecretsRequest")
+    @ResponsePayload
+    public StoreSecretsResponse storeSecretsRequest(@RequestPayload StoreSecretsRequest request) throws OsgpException {
+
+        log.info("Handling incoming SOAP request 'storeSecretsRequest' for device {}", request.getDeviceId());
+        log.trace(request.toString());
+
+        StoreSecretsResponse response = new StoreSecretsResponse();
+
+        try {
+            TypedSecrets soapTypedSecrets = request.getTypedSecrets();
+
+            if (soapTypedSecrets == null) {
+                throw new TechnicalException("Missing input: typed secrets");
+            }
+
+            List<TypedSecret> typedSecretList = converter.convertToTypedSecrets(request.getTypedSecrets());
+
+            secretManagementService.storeSecrets(request.getDeviceId(), typedSecretList);
+
+            response.setResult(OsgpResultType.OK);
+
+            log.trace(response.toString());
+
+            return response;
+        } catch (Exception e) {
+            throw new TechnicalException(ComponentType.SHARED, e.getMessage());
+        }
+    }
+
+    private String getSecretsRequestToString(GetSecretsRequest request) {
+        try {
+            JAXBContext ctx = JAXBContext.newInstance(GetSecretsRequest.class);
+            Marshaller marshaller = ctx.createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            marshaller.marshal(request, baos);
+
+            return baos.toString();
+        }
+        catch(JAXBException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)